src/3d/terrain/qgsterraintexturegenerator_p.cpp

```diff
@@ -159,6 +159,7 @@ QgsMapSettings QgsTerrainTextureGenerator::baseMapSettings()
   mapSettings.setFlag( Qgis::MapSettingsFlag::Render3DMap );
   mapSettings.setTransformContext( mMap.transformContext() );
   mapSettings.setPathResolver( mMap.pathResolver() );
+  mapSettings.setRendererUsage( Qgis::RendererUsage::View );
```

Comment:
What about when a 3d map is being exported via layouts?

Reply:
very good point, that was not treated...
Fixed with https://github.com/qgis/QGIS/pull/46731/commits/5a5bd20fde24867eca9b7c4d15ee56df0071379a but I don't like that we need to reset the engine to force to render again.